fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/FieldKeyExpression.java

```diff
@@ -315,6 +315,29 @@ public SplitKeyExpression split(int splitSize) {
         return field.getMessageType();
     }
 
+    /**
+     * Only supporting DOUBLE, INT64, INT32, and BOOL currently.
+     *
+     * @param descriptor Record Descriptor
+     * @return boolean
+     */
+    @Override
+    public boolean isFixedWidth(Descriptors.Descriptor descriptor) {
+        Descriptors.FieldDescriptor fieldDescriptor = descriptor.findFieldByName(fieldName);
+        if (fieldDescriptor.isRepeated() || fieldDescriptor.isOptional()) {
+            return false;
+        }
+        switch (fieldDescriptor.getType()) {
```

Comment:
I think you could probably switch on `getJavaType` here and then this would also work for a few more numeric types, like `SFIXED32`, `SINT64`, etc.

Reply:
Agreed, changing...